### PR TITLE
use temp folder to push helm charts

### PR DESCRIPTION
**Pull request by Madhu-1**
install helm directly from the script instead of storing it in a temp file and then executing it.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

fixes https://github.com/ceph/ceph-csi/issues/980

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -89,9 +89,7 @@ if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then
 	mkdir -p tmp
```
Comment (nixpanic) [inline deploy.sh:89]: To really address the linked issue, use
```
CSI_CHARTS_DIR=$(mktemp -d)
pushd ${CSI_CHARTS_DIR}
...
popd
...
push_helm_charts... and make sure it uses ${CSI_CHARTS_DIR}
[ -n "${CSI_CHARTS_DIR}" ] && rm -rf ${CSI_CHARTS_DIR}
```

**Comment by humblec** [comment]
@Madhu-1 Is `deploy.sh` meant to be used in a local system? if yes, keeping it in a seperate `tmp` folder is preferred.

**Comment by Madhu-1** [comment]
> @Madhu-1 Is `deploy.sh` meant to be used in a local system? if yes, keeping it in a seperate `tmp` folder is preferred.

not meant to be used, but still, we need it as sometimes we need to test changes locally.

**Comment by Madhu-1** [comment]
@nixpanic @humblec PTAL

**Comment by humblec** [comment]
> > @Madhu-1 Is `deploy.sh` meant to be used in a local system? if yes, keeping it in a seperate `tmp` folder is preferred.
> 
> not meant to be used, but still, we need it as sometimes we need to test changes locally.

Yeah, in those case the `get` helm  inside a `tmp` like folder helps to idenity or not to mangle existing  copy or easy path to wipe as well. Just some thoughts. Its easy to keep everything in a `tmp` folder then remove before next run compared to be  in the `root` dir or `PWD`.

File: deploy.sh [hunk 2]
```diff
@@ -20,13 +21,14 @@ push_helm_charts() {
...
 	fi
 
-	mkdir -p tmp/csi-charts/docs/"$PACKAGE"
-	pushd tmp/csi-charts/docs/"$PACKAGE" >/dev/null
+	mkdir -p "$CHARTDIR/csi-charts/docs/$PACKAGE"
+	cp -R ./charts/ceph-csi-"$PACKAGE" "$CHARTDIR/csi-charts/docs/$PACKAGE"
+	pushd "$CHARTDIR/csi-charts/docs/$PACKAGE" >/dev/null
 	helm init --client-only
-	helm package ../../../../charts/ceph-csi-"$PACKAGE"
+	helm package "$CHARTDIR/csi-charts/docs/$PACKAGE/ceph-csi-$PACKAGE"
```
Comment (nixpanic) [inline deploy.sh:28]: you go into the `"$CHARTDIR/csi-charts/docs/$PACKAGE"` directory, and then you specify the full path here again. Is that really needed?

File: deploy.sh [hunk 3]
```diff
@@ -20,13 +21,14 @@ push_helm_charts() {
...
 	popd >/dev/null
 
-	pushd tmp/csi-charts/docs >/dev/null
+	pushd "$CHARTDIR"/csi-charts/docs >/dev/null
```
Comment (nixpanic) [inline deploy.sh:31]: inconsistent use of `"..."` quoting, only quote `$CHARTDIR` everywhere, or the full path

Comment (Madhu-1) [reply deploy.sh:31]: done

File: deploy.sh (hunk 2 above)
Comment (Madhu-1) [reply deploy.sh]: this also works, however, removed the complete path

**Comment by Madhu-1** [comment]
@humblec @nixpanic PTAL

**Comment by Madhu-1** [comment]
@humblec PTAL

**Comment by Madhu-1** [comment]
@humblec PTAL

**Comment by humblec** [comment]
@Madhu-1 can you quickly rebase this PR ? LGTM.

**Comment by Madhu-1** [comment]
> @Madhu-1 can you quickly rebase this PR ? LGTM.

already rebased

**Pull request merged**